python/GafferSceneUI/PrimitiveVariableQueryUI.py

```diff
@@ -0,0 +1,515 @@
+		self.__row[0].append(
+			nameWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		existsLabelWidget = GafferUI.LabelPlugValueWidget(
+			{ plug["exists"] for plug in self.getPlugs() },
+			horizontalAlignment = GafferUI.Label.HorizontalAlignment.Right
+		)
+		existsLabelWidget.label()._qtWidget().setFixedWidth( 40 )
+		self.__row[0].append(
+			existsLabelWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		self.__row[0].append(
+			GafferUI.BoolPlugValueWidget( { plug["exists"] for plug in self.getPlugs() } ),
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		typeLabelWidget = GafferUI.LabelPlugValueWidget(
+			{ plug['type'] for plug in self.getPlugs() },
+			horizontalAlignment = GafferUI.Label.HorizontalAlignment.Right
+		)
+		typeLabelWidget.label()._qtWidget().setFixedWidth( 30 )
+		self.__row[0].append(
+			typeLabelWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		self.__row[0].append(
+			GafferUI.StringPlugValueWidget( { plug["type"] for plug in self.getPlugs() } ),
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		interpLabelWidget = GafferUI.LabelPlugValueWidget(
+			{ plug["interpolation"] for plug in self.getPlugs() },
+			horizontalAlignment = GafferUI.Label.HorizontalAlignment.Right
+		)
+		interpLabelWidget.label()._qtWidget().setFixedWidth( 80 )
+		self.__row[0].append(
+			interpLabelWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+		interpPresetWidget = GafferUI.PresetsPlugValueWidget(
+			{ plug["interpolation"] for plug in self.getPlugs() } )
+		interpPresetWidget.menuButton()._qtWidget().setFixedWidth( 90 )
+		self.__row[0].append( interpPresetWidget )
+
+		self.__row[1].append(
+			GafferUI.Spacer(
+				imath.V2i( GafferUI.PlugWidget.labelWidth(), 0 ),
+				imath.V2i( GafferUI.PlugWidget.labelWidth(), 0 )
+			),
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		self.__row[1].append(
+			GafferUI.PlugValueWidget.create( { plug["value"] for plug in self.getPlugs() } )
+		)
+
+	def setPlugs( self, plugs ) :
+
+		GafferUI.PlugValueWidget.setPlugs( self, plugs )
+
+		self.__row[0][0].setPlugs( plugs )
+		self.__row[0][1].setPlugs( { plug["exists"] for plug in plugs } )
+		self.__row[0][2].setPlugs( { plug["exists"] for plug in plugs } )
+		self.__row[0][3].setPlugs( { plug["type"] for plug in plugs } )
+		self.__row[0][4].setPlugs( { plug["type"] for plug in plugs } )
+		self.__row[0][5].setPlugs( { plug["interpolation"] for plug in plugs } )
+		self.__row[0][6].setPlugs( { plug["interpolation"] for plug in plugs } )
+		self.__row[1][1].setPlugs( { plug["value"] for plug in plugs } )
+
+	def hasLabel( self ) :
+
+		return True
+
+	def childPlugValueWidget( self, childPlug ) :
+
+		for row in self.__row :
+			for w in row :
+				if childPlug in w.getPlugs() :
+					return w
+
+		return None
+
+	def setReadOnly( self, readOnly ) :
```

Comment:
Note to self : this method is removed in #4970, so I might need to remove this on merging to `main`.